indexing-service/src/main/java/io/druid/indexing/common/task/HadoopTask.java

```diff
@@ -93,7 +93,7 @@ protected ClassLoader buildClassLoader(final TaskToolbox toolbox) throws Excepti
       driverURLs.addAll(Arrays.asList(((URLClassLoader) hadoopLoader).getURLs()));
     }
 
-    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), null);
+    final URLClassLoader loader = new URLClassLoader(driverURLs.toArray(new URL[driverURLs.size()]), Thread.currentThread().getContextClassLoader());
```

Comment:
This might need to be in its own PR as it looks like a classloader bug independent of this PR. (or it can wait for this one)